config-handler-router/src/test/groovy/org/codice/admin/router/ConfigurationHandlerRouterTest.groovy

```diff
@@ -0,0 +1,294 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.admin.router
+
+import org.codice.ddf.admin.api.config.Configuration
+import org.codice.ddf.admin.api.config.ConfigurationType
+import org.codice.ddf.admin.api.config.sources.WfsSourceConfiguration
+import org.codice.ddf.admin.api.handler.ConfigurationHandler
+import org.codice.ddf.admin.api.handler.report.CapabilitiesReport
+import org.codice.ddf.admin.api.handler.report.ProbeReport
+import org.codice.ddf.admin.api.handler.report.Report
+import spark.Request
+import spark.Response
+import spock.lang.Specification
+
+class ConfigurationHandlerRouterTest extends Specification {
+    
+    static final FAILURE_MESSAGE = String.format(ConfigurationHandlerRouter.NO_CONFIG_ID_MSG, NON_MATCHING_ID)
+
+    static final CONFIGURATION_TYPE = "configurationType"
+
+    static final MATCHING_ID = "myId"
+
+    static final NON_MATCHING_ID = "notMyId"
+
+    static final METHOD_ID = "someId"
+
+    ConfigurationHandlerRouter configurationHandlerRouter
+
+    Response response
+
+    ConfigurationHandler configurationHandler
+
+    Request request
+
+    def setup() {
+        configurationHandlerRouter = new ConfigurationHandlerRouter()
+    }
+
+    def 'test test(Request, Response) config handler success'() {
+        setup:
+        prepareConfigHandler(MATCHING_ID, ConfigurationHandlerRouter.TEST_ID)
+
+        def report = mockReport(false)
+
+        when:
+        configurationHandlerRouter.test(request, response)
+
+
+        then:
+        1 * configurationHandler.test(METHOD_ID, _ as Configuration) >> report
```

Comment:
Can we also check the subtype of the messages returned in all these tests too please